src/context/runtime.go

```diff
@@ -10,30 +10,71 @@ import (
 var Current *Runtime
 
 type Runtime struct {
-	Shell string
-	OS    string
-	Home  string
-	Arch  string
-	Path  *Path
+	Shell     string
+	OS        string
+	Home      string
+	ConfigDir string
+	CacheDir  string
+	Arch      string
+	User      string
+	Path      *Path
+	Hostname  string
 }
 
 func Init(shell string) {
 	Current = &Runtime{
-		Shell: shell,
-		OS:    runtime.GOOS,
-		Arch:  runtime.GOARCH,
-		Home:  Home(),
-		Path:  getPath(),
+		Shell:     shell,
+		OS:        runtime.GOOS,
+		Arch:      runtime.GOARCH,
+		User:      os.Getenv("USER"),
+		Home:      Home(),
+		ConfigDir: ConfigDir(),
+		CacheDir:  CacheDir(),
+		Path:      getPath(),
+		Hostname:  Hostname(),
 	}
 }
 
+func ConfigDir() string {
+	if Current != nil {
+		return Current.ConfigDir
+	}
+	path, err := os.UserConfigDir()
+	if err == nil && len(path) > 0 {
+		return path
+	}
+	return ""
+}
+
+func CacheDir() string {
+	if Current != nil {
+		return Current.CacheDir
+	}
+	path, err := os.UserCacheDir()
+	if err == nil && len(path) > 0 {
+		return path
+	}
+	return ""
+}
+
+func Hostname() string {
+	if Current != nil {
+		return Current.Hostname
+	}
+	hostname, err := os.Hostname()
+	if err == nil && len(hostname) > 0 {
```

Comment:
`len(hostname) > 0` is obsolete as we return `""` anyways on error. So I don't see a change in behaviour.